src/Altinn.Notifications.Email/Attributes/AccessKeyAttribute.cs

```diff
@@ -0,0 +1,29 @@
+﻿using Altinn.Notifications.Email.Configuration;
+
+using Microsoft.AspNetCore.Mvc.Filters;
+
+namespace Altinn.Notifications.Email.Attributes;
+
+/// <summary>
+/// Attribute for marking a controller or action that requires an access key for authorization.
+/// </summary>
+[AttributeUsage(AttributeTargets.Class | AttributeTargets.Method)]
+public class AccessKeyAttribute : Attribute, IAsyncActionFilter
+{
+    /// <summary>
+    /// Method that is called before the action is executed.
+    /// </summary>
+    public async Task OnActionExecutionAsync(ActionExecutingContext context, ActionExecutionDelegate next)
+    {
+        AuthSettings authSettings = context.HttpContext.RequestServices.GetRequiredService<IConfiguration>()!.GetSection("AuthSettings").Get<AuthSettings>()!;
```

Comment:
We may need a longer section name to match in the environment vars. AuthSettings__Accesskey is a secret name that might drown a bit in key vault amongst all other secrets?